src/java.base/share/classes/java/net/InetSocketAddress.java

```diff
@@ -373,11 +373,18 @@ public final boolean isUnresolved() {
 
     /**
      * Constructs a string representation of this InetSocketAddress.
-     * This String is constructed by calling toString() on the InetAddress
-     * and concatenating the port number (with a colon). If the address
-     * is an IPv6 address, the IPv6 literal is enclosed in square brackets.
+     * This string is constructed by calling {@link InetAddress#toString()}
+     * on the InetAddress and concatenating the port number (with a colon).
+     * <p>
+     * If the address is an IPv6 address, the IPv6 literal is enclosed in
+     * square brackets, for example: {@code "localhost/[0:0:0:0:0:0:0:1]:80"}.
      * If the address is {@linkplain #isUnresolved() unresolved},
-     * {@code <unresolved>} is displayed in place of the address literal.
+     * {@code <unresolved>} is displayed in place of the address literal, for
+     * example {@code "foo/<unresolved>:80"}.
+     * <p>
+     * To retrieve a string representation of the hostname or the address, use
+     * {@link #getHostString()}, rather than parsing the string returned by this
+     * {@link #toString()} method.
```

Comment:
A couple of small suggestions:
- "This string" -> "The string"
- For the IPv6 literal case it would good to make it clear that it is enclosed in square brackets before append the colon and port. It might also be better to not have this in a separate paragraph because it follows immediately from the previous sentence.
- The final sentence "To retrieve ..." might be better starting with "Use getHostString to get the String representation ..."

Reply:
Agreed offline to not apply these suggestions